src/languages/rust.ts

```diff
@@ -0,0 +1,175 @@
+import { SyntaxNode } from "web-tree-sitter";
+import { SimpleScopeTypeType } from "../typings/targetDescriptor.types";
+import { NodeMatcherAlternative } from "../typings/Types";
+import { patternFinder } from "../util/nodeFinders";
+import {
+  ancestorChainNodeMatcher,
+  argumentMatcher,
+  cascadingMatcher,
+  createPatternMatchers,
+  leadingMatcher,
+  matcher,
+  patternMatcher,
+  trailingMatcher,
+} from "../util/nodeMatchers";
+
+// Generated by the following command:
+// `curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-rust/36ae187ed6dd3803a8a89dbb54f3124c8ee74662/src/node-types.STATEMENT_TYPES | jq '[.[] | select(.type == "_declaration_statement") | .subtypes[].type, "expression_statement"]'`
+const STATEMENT_TYPES = [
+  "associated_type",
+  "attribute_item",
+  "const_item",
+  "empty_statement",
+  "enum_item",
+  "extern_crate_declaration",
+  "foreign_mod_item",
+  "impl_item",
+  "inner_attribute_item",
+  "let_declaration",
+  "macro_definition",
+  "macro_invocation",
+  "function_item",
+  "function_signature_item",
+  "mod_item",
+  "static_item",
+  "struct_item",
+  "trait_item",
+  "type_item",
+  "union_item",
+  "use_declaration",
+  "expression_statement",
+];
+
+/**
+ * Scope types allowed to be parents of a statement
+ */
+const STATEMENT_PARENT_TYPES = ["source_file", "block", "declaration_list"];
+
+/**
+ * Returns "impl_item[type]" node higher in the chain
+ * @param node The node which we will start our search from
+ * @returns node or null
+ */
+function traitTypeMatcher(node: SyntaxNode) {
```

Comment:
This name isn't really accurate, as you're matching impl item.  But you probably also want to match trait, right?  Maybe just ignore the parent node type entirely and check if this node is at the `type` field of its parent?  Then just call this function `typeFieldFinder` or something.  

Could prob just have this be a generic matcher, where it takes a field name, and returns a matcher that looks for a node that appears as that field

Reply:
> This name isn't really accurate, as you're matching impl item

I agree, I’ll think about it some more.

> But you probably also want to match trait, right?

I could do that, but I don’t have to. A simple matcher for “impl_item” does that already.

> Maybe just ignore the parent node type entirely and check if this node is at the type field of its parent? Then just call this function typeFieldFinder or something

We can certainly make it more generic. But I would argue that there should be two arguments to this function the parent type in a chain and its child’s type. It should match if we are anywhere inside specific subtree in parse tree.

Reply:
Ah right I guess you don't need it for `trait_item`.  I guess fine to just rename the function to `implItemTypeFinder` and leave the more generic thing for now